modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/repo/es/ElasticSearchDetectorMappingService.java

```diff
@@ -0,0 +1,377 @@
+                elasticSearchConfig.getDocType(), index);
+        indexRequest.source(DetectorMappingEntityToJson(detectorMappingEntity), XContentType.JSON);
+        try {
+            elasticSearchClient.index(indexRequest, RequestOptions.DEFAULT);
+        } catch (IOException e) {
+            log.error("Update Index failed", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void deleteDetectorMapping(String id) {
+        final DeleteRequest deleteRequest = new DeleteRequest(elasticSearchConfig.getIndexName(),
+                elasticSearchConfig.getDocType(), id);
+        try {
+            elasticSearchClient.delete(deleteRequest, RequestOptions.DEFAULT);
+        } catch (IOException e) {
+            log.error(String.format("Deleting mapping %s failed", id), e);
+            throw new RuntimeException(e);
+        }
+    }
+    
+    @Override
+    public void disableDetectorMapping(String id) {
+        DetectorMapping detectorMapping = findDetectorMapping(id);
+        if (detectorMapping.isEnabled()) {
+            final DetectorMappingEntity detectorMappingEntity = new DetectorMappingEntity(detectorMapping.getUser(),
+                    detectorMapping.getDetector(), QueryUtil.buildQuery(detectorMapping.getExpression()),
+                    false,
+                    System.currentTimeMillis(), detectorMapping.getCreatedTimeInMillis());
+            updateDetectorMapping(id, detectorMappingEntity);
+        }
+    }
+
+    @Override
+    public DetectorMapping findDetectorMapping(String id) {
+        GetRequest getRequest = new GetRequest(elasticSearchConfig.getIndexName(),
+            elasticSearchConfig.getDocType(), id);
+        try {
+            GetResponse response = elasticSearchClient.get(getRequest, RequestOptions.DEFAULT);
+            return getDetectorMapping(response.getSourceAsString(), response.getId(), Optional.empty());
+        } catch (IOException e) {
+            log.error(String.format("Get mapping %s failed", id), e);
+            throw new RuntimeException(e);
+        }
+    }
+    
+    @Override
+    public List<DetectorMapping> search(SearchMappingsRequest searchMappingsRequest) {
+        SearchRequest searchRequest = new SearchRequest(elasticSearchConfig.getIndexName());
+        searchRequest.types(elasticSearchConfig.getDocType());
+        BoolQueryBuilder query = QueryBuilders.boolQuery();
+        if (searchMappingsRequest.getUserId() != null) {
+            query.must(userIdQuery(searchMappingsRequest));
+        }
+        if (searchMappingsRequest.getDetectorUuid() != null) {
+            query.must(detectorIdQuery(searchMappingsRequest));
+        }
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.query(query);
+        //FIXME setting default result set size to 10000 until we have pagination.
+        searchSourceBuilder.size(10000);
```

Comment:
Couple concerns:

- If neither the `userId` nor the `detectorUuid` are specified, then this ends up returning all mappings (or at least 10,000 mappings).
- 10,000 is too large of a result set, even for a temporary situation.

I'd request the following:

- Change 10,000 to 500. We can make this paging when we adopt Spring Data ES, but until then we should limit resource consumption.
- Get rid of `SearchMappingsRequest`. Instead turn this method into two separate methods: `findByUserId` and `findByDetectorUuid`. I'm assuming there's no use case for finding by both jointly since the client could always find by detector UUID and then check the user ID if needed.

Reply:
Updated to 500, will review the use case for finding by both as a separate change.